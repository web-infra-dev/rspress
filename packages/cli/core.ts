export * from '@rspress/core';

export type { RspressPlugin } from '@rspress/shared';

export { logger } from '@rspress/shared/logger';
