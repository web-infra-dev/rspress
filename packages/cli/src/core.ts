export * from '@rspress/core/dist/index.js';
