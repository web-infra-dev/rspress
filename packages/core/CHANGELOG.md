# @rspress/core

## 1.5.1

### Patch Changes

- d616c0b: chore(deps): bump Rsbuild v0.0.14 and remove Modern.js plugin testing
- 5b413ec: fix: HomeLayout cannot be customized
- a852f65: perf: using polyfill usage mode to reduce bundle size
- 5b413ec: fix: sidebar disappear when the `link` of items doesn't start with the sidebar group key
- Updated dependencies [d616c0b]
- Updated dependencies [5b413ec]
- Updated dependencies [fa91446]
- Updated dependencies [5b413ec]
  - @rspress/shared@1.5.1
  - @rspress/theme-default@1.5.1
  - @rspress/plugin-auto-nav-sidebar@1.5.1
  - @rspress/plugin-container-syntax@1.5.1
  - @rspress/plugin-last-updated@1.5.1
  - @rspress/plugin-medium-zoom@1.5.1
  - @rspress/runtime@1.5.1

## 1.5.0

### Minor Changes

- 1165c9a: feat: bash syntax highlighting is supported by default
- 8043517: feat: add support for wrap code in markdown code block

### Patch Changes

- 750ed07: feat(deps): bump Rsbuild from 0.0.10 to 0.0.13
- Updated dependencies [f9fbc46]
- Updated dependencies [29f3dbd]
- Updated dependencies [750ed07]
- Updated dependencies [5bcb36c]
- Updated dependencies [1b6e6e3]
- Updated dependencies [8043517]
  - @rspress/theme-default@1.5.0
  - @rspress/shared@1.5.0
  - @rspress/plugin-auto-nav-sidebar@1.5.0
  - @rspress/plugin-container-syntax@1.5.0
  - @rspress/plugin-last-updated@1.5.0
  - @rspress/plugin-medium-zoom@1.5.0
  - @rspress/runtime@1.5.0

## 1.4.1

### Patch Changes

- d7fec5f: fix: tsx file in node_modules cannot be compiled
- 9659d65: fix: use theme alias path
  - @rspress/plugin-auto-nav-sidebar@1.4.1
  - @rspress/plugin-container-syntax@1.4.1
  - @rspress/plugin-last-updated@1.4.1
  - @rspress/plugin-medium-zoom@1.4.1
  - @rspress/runtime@1.4.1
  - @rspress/shared@1.4.1
  - @rspress/theme-default@1.4.1

## 1.4.0

### Patch Changes

- @rspress/plugin-auto-nav-sidebar@1.4.0
- @rspress/plugin-container-syntax@1.4.0
- @rspress/plugin-last-updated@1.4.0
- @rspress/plugin-medium-zoom@1.4.0
- @rspress/runtime@1.4.0
- @rspress/shared@1.4.0
- @rspress/theme-default@1.4.0

## 1.3.2

### Patch Changes

- Updated dependencies [d3b08e7]
  - @rspress/theme-default@1.3.2
  - @rspress/runtime@1.3.2
  - @rspress/plugin-medium-zoom@1.3.2
  - @rspress/plugin-auto-nav-sidebar@1.3.2
  - @rspress/plugin-container-syntax@1.3.2
  - @rspress/plugin-last-updated@1.3.2
  - @rspress/shared@1.3.2

## 1.3.1

### Patch Changes

- 91c35f6: fix: @theme alias not found
  - @rspress/plugin-auto-nav-sidebar@1.3.1
  - @rspress/plugin-container-syntax@1.3.1
  - @rspress/plugin-last-updated@1.3.1
  - @rspress/plugin-medium-zoom@1.3.1
  - @rspress/runtime@1.3.1
  - @rspress/shared@1.3.1
  - @rspress/theme-default@1.3.1

## 1.3.0

### Minor Changes

- c71b487: feat: add support for the dev port or host param
- 9aa5bbf: feat: add support HTML to frontmatter hero config
- 7dd0103: feat: support home feature grid 3xN & costom grid length

### Patch Changes

- 9c8b9f9: feat: integrate Rsbuild to replace Modern.js Builder
- 4a6a052: feat(core): beautify <Tab> sytle when using with code title on default theme
- 42f982d: fix: code tag in nested jsx structure cannot be parsed in preview/playground plugin
- a267cc9: fix: theme types missed
- Updated dependencies [0ff27fd]
- Updated dependencies [c5b864f]
- Updated dependencies [500e5bf]
- Updated dependencies [d66037f]
- Updated dependencies [9c8b9f9]
- Updated dependencies [daaf70d]
- Updated dependencies [bc17b81]
- Updated dependencies [d3eaa8f]
- Updated dependencies [7dd0103]
- Updated dependencies [1e4827c]
  - @rspress/theme-default@1.3.0
  - @rspress/shared@1.3.0
  - @rspress/plugin-auto-nav-sidebar@1.3.0
  - @rspress/plugin-last-updated@1.3.0
  - @rspress/plugin-medium-zoom@1.3.0
  - @rspress/runtime@1.3.0
  - @rspress/plugin-container-syntax@1.3.0

## 1.2.1

### Patch Changes

- b8226b0: fix: rspress-doc-container margin css style
  - @rspress/plugin-auto-nav-sidebar@1.2.1
  - @rspress/plugin-container-syntax@1.2.1
  - @rspress/plugin-last-updated@1.2.1
  - @rspress/plugin-medium-zoom@1.2.1
  - @rspress/shared@1.2.1

## 1.2.0

### Minor Changes

- 0687d48: feat: add runtime module hook
- d57a88c: feat: support ssg config

### Patch Changes

- 1575a79: fix: anchor missed when link exists in header
- 8e64681: fix social link message and image display bug
- 4143291: fix(core): Fix background of HiddenLinks to support dark theme
- ac665cd: fix: production error for preview plugin
- 29698ed: fix: app crashed when creating \_meta.json or appearing frontmatter syntax error
- a5b9678: feat: support content in overview component
- Updated dependencies [0687d48]
- Updated dependencies [d57a88c]
- Updated dependencies [31e50ed]
  - @rspress/shared@1.2.0
  - @rspress/plugin-auto-nav-sidebar@1.2.0
  - @rspress/plugin-container-syntax@1.2.0
  - @rspress/plugin-last-updated@1.2.0
  - @rspress/plugin-medium-zoom@1.2.0

## 1.1.3

### Patch Changes

- a1c999f: fix: outline hidden style when headers is empty
  - @rspress/plugin-auto-nav-sidebar@1.1.3
  - @rspress/plugin-container-syntax@1.1.3
  - @rspress/plugin-last-updated@1.1.3
  - @rspress/plugin-medium-zoom@1.1.3
  - @rspress/shared@1.1.3

## 1.1.2

### Patch Changes

- @rspress/plugin-auto-nav-sidebar@1.1.2
- @rspress/plugin-container-syntax@1.1.2
- @rspress/plugin-last-updated@1.1.2
- @rspress/plugin-medium-zoom@1.1.2
- @rspress/shared@1.1.2

## 1.1.1

### Patch Changes

- adc6924: fix: resolve error for react-syntax-highlighter
  - @rspress/plugin-auto-nav-sidebar@1.1.1
  - @rspress/plugin-container-syntax@1.1.1
  - @rspress/plugin-last-updated@1.1.1
  - @rspress/plugin-medium-zoom@1.1.1
  - @rspress/shared@1.1.1

## 1.1.0

### Minor Changes

- c9c894d: feat: support url query to adjust page layout
- 342c932: feat: add configure prism languages

### Patch Changes

- 887b0cb: feat: support multi version menu in nav
- c4ed0e7: fix(style): keep overview h1 size same as other pages
- 6cfc67e: perf(core): reduce polyfills by adjusting browserslist
- 6732716: feat: set metakey `ctrl` in windows
- b1fe901: fix(style): limit the max width of hero tagline
- f0dfa65: feat: support remove plugin in config hook
- 8c1dce1: feat: support cutsom anchor in default mdx compiler
- Updated dependencies [887b0cb]
- Updated dependencies [ff42bc6]
- Updated dependencies [0e8d366]
- Updated dependencies [342c932]
  - @rspress/shared@1.1.0
  - @rspress/plugin-auto-nav-sidebar@1.1.0
  - @rspress/plugin-container-syntax@1.1.0
  - @rspress/plugin-last-updated@1.1.0
  - @rspress/plugin-medium-zoom@1.1.0

## 1.0.2

### Patch Changes

- a59f05c: chore: optimize readme in npm
- Updated dependencies [a59f05c]
  - @rspress/plugin-auto-nav-sidebar@1.0.2
  - @rspress/plugin-container-syntax@1.0.2
  - @rspress/plugin-last-updated@1.0.2
  - @rspress/plugin-medium-zoom@1.0.2
  - @rspress/shared@1.0.2

## 1.0.1

### Patch Changes

- @rspress/plugin-auto-nav-sidebar@1.0.1
- @rspress/plugin-container-syntax@1.0.1
- @rspress/plugin-last-updated@1.0.1
- @rspress/plugin-medium-zoom@1.0.1
- @rspress/shared@1.0.1

## 1.0.0

### Minor Changes

- 213c9e3: fix: tuning the animation for `first entry` and `navigate with hash`
- d1098cf: feat: support cleanUrls to generate url without suffix

### Patch Changes

- 7f771b8: feat: add `markdown.mdxRs` config
- 599291c: fix: dead link invalid when having a trailing slash

  fix: 路径包含尾部斜杠时，死链检测失效

- 4a133ab: feat: support tabPosition in Tabs
- 34ac6ed: chore: modify search hooks default extension
- fd49190: feat: make navbar flexible
- 3b89c1d: feat: more flexible root config
- 1851d12: fix: home hero image path error when base path exists
- 12089b7: feat: support hideNavbar config
- a9271e9: feat: using rslog as logger

  feat: 使用 rslog 作为 logger

- 12d9569: chore: remove unuse peer dep
- 485d873: fix: home page hmr invalid

  fix: home 页面热更新失效

- Updated dependencies [fdf66e4]
- Updated dependencies [d1098cf]
- Updated dependencies [a9271e9]
  - @rspress/plugin-medium-zoom@1.0.0
  - @rspress/shared@1.0.0
  - @rspress/plugin-auto-nav-sidebar@1.0.0
  - @rspress/plugin-container-syntax@1.0.0
  - @rspress/plugin-last-updated@1.0.0

## 1.0.0-beta.5

### Minor Changes

- 213c9e3: fix: tuning the animation for `first entry` and `navigate with hash`
- d1098cf: feat: support cleanUrls to generate url without suffix

### Patch Changes

- 7f771b8: feat: add `markdown.mdxRs` config
- 599291c: fix: dead link invalid when having a trailing slash

  fix: 路径包含尾部斜杠时，死链检测失效

- 4a133ab: feat: support tabPosition in Tabs
- 34ac6ed: chore: modify search hooks default extension
- fd49190: feat: make navbar flexible
- 3b89c1d: feat: more flexible root config
- 1851d12: fix: home hero image path error when base path exists
- 12089b7: feat: support hideNavbar config
- a9271e9: feat: using rslog as logger

  feat: 使用 rslog 作为 logger

- 12d9569: chore: remove unuse peer dep
- 485d873: fix: home page hmr invalid

  fix: home 页面热更新失效

- Updated dependencies [fdf66e4]
- Updated dependencies [d1098cf]
- Updated dependencies [a9271e9]
  - @rspress/plugin-medium-zoom@1.0.0-beta.5
  - @rspress/shared@1.0.0-beta.5
  - @rspress/plugin-auto-nav-sidebar@1.0.0-beta.5
  - @rspress/plugin-container-syntax@1.0.0-beta.5
  - @rspress/plugin-last-updated@1.0.0-beta.5

## 1.0.0-beta.4

### Minor Changes

- 213c9e3: fix: tuning the animation for `first entry` and `navigate with hash`
- d1098cf: feat: support cleanUrls to generate url without suffix

### Patch Changes

- 7f771b8: feat: add `markdown.mdxRs` config
- 599291c: fix: dead link invalid when having a trailing slash

  fix: 路径包含尾部斜杠时，死链检测失效

- 4a133ab: feat: support tabPosition in Tabs
- 34ac6ed: chore: modify search hooks default extension
- fd49190: feat: make navbar flexible
- 3b89c1d: feat: more flexible root config
- 1851d12: fix: home hero image path error when base path exists
- 12089b7: feat: support hideNavbar config
- a9271e9: feat: using rslog as logger

  feat: 使用 rslog 作为 logger

- 12d9569: chore: remove unuse peer dep
- 485d873: fix: home page hmr invalid

  fix: home 页面热更新失效

- Updated dependencies [d1098cf]
- Updated dependencies [a9271e9]
  - @rspress/shared@1.0.0-beta.4
  - @rspress/plugin-auto-nav-sidebar@1.0.0-beta.4
  - @rspress/plugin-container-syntax@1.0.0-beta.4
  - @rspress/plugin-last-updated@1.0.0-beta.4
  - @rspress/plugin-medium-zoom@1.0.0-beta.4

## 1.0.0-beta.3

### Minor Changes

- 213c9e3: fix: tuning the animation for `first entry` and `navigate with hash`
- d1098cf: feat: support cleanUrls to generate url without suffix

### Patch Changes

- 7f771b8: feat: add `markdown.mdxRs` config
- 599291c: fix: dead link invalid when having a trailing slash

  fix: 路径包含尾部斜杠时，死链检测失效

- 4a133ab: feat: support tabPosition in Tabs
- 34ac6ed: chore: modify search hooks default extension
- 3b89c1d: feat: more flexible root config
- a9271e9: feat: using rslog as logger

  feat: 使用 rslog 作为 logger

- 12d9569: chore: remove unuse peer dep
- 485d873: fix: home page hmr invalid

  fix: home 页面热更新失效

- Updated dependencies [d1098cf]
- Updated dependencies [a9271e9]
  - @rspress/shared@1.0.0-beta.3
  - @rspress/plugin-auto-nav-sidebar@1.0.0-beta.3
  - @rspress/plugin-container-syntax@1.0.0-beta.3
  - @rspress/plugin-last-updated@1.0.0-beta.3
  - @rspress/plugin-medium-zoom@1.0.0-beta.3

## 1.0.0-beta.2

### Minor Changes

- 213c9e3: fix: tuning the animation for `first entry` and `navigate with hash`
- d1098cf: feat: support cleanUrls to generate url without suffix

### Patch Changes

- 599291c: fix: dead link invalid when having a trailing slash

  fix: 路径包含尾部斜杠时，死链检测失效

- 4a133ab: feat: support tabPosition in Tabs
- 34ac6ed: chore: modify search hooks default extension
- 3b89c1d: feat: more flexible root config
- a9271e9: feat: using rslog as logger

  feat: 使用 rslog 作为 logger

- 485d873: fix: home page hmr invalid

  fix: home 页面热更新失效

- Updated dependencies [d1098cf]
- Updated dependencies [a9271e9]
  - @rspress/shared@1.0.0-beta.2
  - @rspress/plugin-auto-nav-sidebar@1.0.0-beta.2
  - @rspress/plugin-container-syntax@1.0.0-beta.2
  - @rspress/plugin-last-updated@1.0.0-beta.2
  - @rspress/plugin-medium-zoom@1.0.0-beta.2

## 1.0.0-beta.1

### Minor Changes

- 213c9e3: fix: tuning the animation for `first entry` and `navigate with hash`
- d1098cf: feat: support cleanUrls to generate url without suffix

### Patch Changes

- 599291c: fix: dead link invalid when having a trailing slash

  fix: 路径包含尾部斜杠时，死链检测失效

- 34ac6ed: chore: modify search hooks default extension
- 3b89c1d: feat: more flexible root config
- a9271e9: feat: using rslog as logger

  feat: 使用 rslog 作为 logger

- 485d873: fix: home page hmr invalid

  fix: home 页面热更新失效

- Updated dependencies [d1098cf]
- Updated dependencies [a9271e9]
  - @rspress/shared@1.0.0-beta.1
  - @rspress/plugin-auto-nav-sidebar@1.0.0-beta.1
  - @rspress/plugin-container-syntax@1.0.0-beta.1
  - @rspress/plugin-last-updated@1.0.0-beta.1
  - @rspress/plugin-medium-zoom@1.0.0-beta.1

## 1.0.0-beta.0

### Minor Changes

- 213c9e3: fix: tuning the animation for `first entry` and `navigate with hash`
- d1098cf: feat: support cleanUrls to generate url without suffix

### Patch Changes

- 599291c: fix: dead link invalid when having a trailing slash

  fix: 路径包含尾部斜杠时，死链检测失效

- 485d873: fix: home page hmr invalid

  fix: home 页面热更新失效

- Updated dependencies [d1098cf]
  - @rspress/shared@1.0.0-beta.0
  - @rspress/plugin-auto-nav-sidebar@1.0.0-beta.0
  - @rspress/plugin-container-syntax@1.0.0-beta.0
  - @rspress/plugin-last-updated@1.0.0-beta.0
  - @rspress/plugin-medium-zoom@1.0.0-beta.0

## 0.2.0

### Minor Changes

- 178ba42: feat: support multi version docs

  feat: 支持文档多版本

- da74868: using view transition to present animation when page navigate

### Patch Changes

- 100e0f6: fix: preview plugin build failed

  fix: 预览插件构建崩溃

- b56c083: feat: add min-height in .rspress-doc

  feat: 给 .rspress-doc 添加 min-height 样式

- Updated dependencies [178ba42]
- Updated dependencies [da74868]
  - @rspress/plugin-auto-nav-sidebar@0.2.0
  - @rspress/shared@0.2.0
  - @rspress/plugin-container-syntax@0.2.0
  - @rspress/plugin-last-updated@0.2.0
  - @rspress/plugin-medium-zoom@0.2.0

## 0.1.1

### Patch Changes

- @rspress/plugin-auto-nav-sidebar@0.1.1
- @rspress/plugin-container-syntax@0.1.1
- @rspress/plugin-last-updated@0.1.1
- @rspress/plugin-medium-zoom@0.1.1
- @rspress/shared@0.1.1

## 0.1.0

### Minor Changes

- 6132722: feat: integrate @rspress/mdx-rs

  feat: 集成 @rspress/mdx-rs

### Patch Changes

- 1b3c4a1: feat: support SearchPanel component export

  feat: 支持 SearchPanel 组件导出

- 05e1b13: fix: runtime type missed

  fix: runtime 目录类型丢失

- e076e31: chore: no need to print the server bundles size

  chore: 无须输出 server bundles 的体积

- f155ee3: fix: container syntax not work

  fix: 容器语法不生效

- db250f2: feat: support icon absolute path

  feat: 支持 icon 配置绝对路径

- Updated dependencies [6132722]
  - @rspress/plugin-auto-nav-sidebar@0.1.0
  - @rspress/plugin-container-syntax@0.1.0
  - @rspress/plugin-last-updated@0.1.0
  - @rspress/plugin-medium-zoom@0.1.0
  - @rspress/shared@0.1.0

## 0.0.13

### Patch Changes

- 80a49c1: feat: remove @headlessui/react dep

  feat: 移除 @headlessui/react 依赖

- 1d42f2a: feat: support PackageManagerTabs export

  feat: 支持 PackageManagerTabs 组件导出

- 038c371: fix: dispaly error log in ssg phase

  fix: ssg 阶段显示错误日志

  - @rspress/plugin-auto-nav-sidebar@0.0.13
  - @rspress/plugin-container-syntax@0.0.13
  - @rspress/plugin-last-updated@0.0.13
  - @rspress/plugin-medium-zoom@0.0.13
  - @rspress/shared@0.0.13

## 0.0.12

### Patch Changes

- Updated dependencies [370b348]
  - @rspress/plugin-auto-nav-sidebar@0.0.12
  - @rspress/plugin-container-syntax@0.0.12
  - @rspress/plugin-last-updated@0.0.12
  - @rspress/plugin-medium-zoom@0.0.12
  - @rspress/shared@0.0.12

## 0.0.11

### Patch Changes

- 47c2e53: fix: remove deprecated npm package

  fix: 移除废弃的 npm 包

- 0896e86: fix: hmr invalid for adding file case

  fix: 增加文件后，热更新失效

- 4e7ca8c: perf: enable deterministic chunkIds of Rspack

  perf: 开启 Rspack 的 deterministic chunkIds

- 5fcc813: fix(mdx-rs): avoid panic when parsing mdx ast
- 0943989: feat: support props for globalUIComponents config

  feat: 支持 `globalUIComponents` 的 props 配置

- Updated dependencies [0896e86]
- Updated dependencies [0943989]
  - @rspress/plugin-auto-nav-sidebar@0.0.11
  - @rspress/plugin-container-syntax@0.0.11
  - @rspress/plugin-last-updated@0.0.11
  - @rspress/plugin-medium-zoom@0.0.11
  - @rspress/shared@0.0.11

## 0.0.10

### Patch Changes

- 82d85e2: feat: integrate rspack-plugin-virtual-module

  feat: 集成 rspack-plugin-virtual-module

- 4ecf59c: chore: update tabs default theme style

  chore: 更新 tabs 默认主题样式

- 0a4dbc6: feat: optimize default theme layout

  feat: 优化默认主题布局

- 259c2e6: fix: avoid error when \_meta.json is not correct

  fix: 修复 \_meta.json 内容不正确时报错

- f6f9ddb: feat: specify limit version for create-rspress template

  feat: 为 create-rspress 模板指定限制 rspress 版本

- 51211bc: fix: asset prefix not work for search index url

  fix: asset prefix 配置在搜索索引中不生效

- 587d299: feat: support container syntax by plugin

  feat: 以插件的方式支持 container 语法

- f21526b: fix: avoid `@theme` resolve error in pure js project

  fix: 纯 js 项目中避免 `@theme` 解析错误

- Updated dependencies [82d85e2]
- Updated dependencies [0a4dbc6]
- Updated dependencies [259c2e6]
- Updated dependencies [f6f9ddb]
- Updated dependencies [51211bc]
- Updated dependencies [587d299]
- Updated dependencies [f21526b]
  - @rspress/plugin-auto-nav-sidebar@0.0.10
  - @rspress/plugin-container-syntax@0.0.10
  - @rspress/plugin-last-updated@0.0.10
  - @rspress/plugin-medium-zoom@0.0.10
  - @rspress/shared@0.0.10

## 0.0.9

### Patch Changes

- b46cdfd: fix: runtime exports problem

  fix: runtime 文件导出问题

- Updated dependencies [b46cdfd]
  - @rspress/plugin-auto-nav-sidebar@0.0.9
  - @rspress/plugin-last-updated@0.0.9
  - @rspress/plugin-medium-zoom@0.0.9
  - @rspress/remark-container@0.0.8
  - @rspress/shared@0.0.9

## 0.0.8

### Patch Changes

- 86677f1: fix: resolve error in pure js project

  fix: 修复纯 js 项目中的路径 resolve 报错

- Updated dependencies [86677f1]
  - @rspress/plugin-auto-nav-sidebar@0.0.8
  - @rspress/plugin-last-updated@0.0.8
  - @rspress/plugin-medium-zoom@0.0.8
  - @rspress/remark-container@0.0.7
  - @rspress/shared@0.0.8

## 0.0.7

### Patch Changes

- 24450c2: feat: export `Plugin` type in @rspress/shared

  feat: 在 @rspress/shared 中导出 `Plugin` 类型

- Updated dependencies [24450c2]
  - @rspress/shared@0.0.7
  - @rspress/plugin-auto-nav-sidebar@0.0.7
  - @rspress/plugin-last-updated@0.0.7
  - @rspress/plugin-medium-zoom@0.0.7
  - @rspress/remark-container@0.0.6

## 0.0.6

### Patch Changes

- beb23d6: fix: landing message in cli

  fix: 更新 cli 的开场信息

- Updated dependencies [beb23d6]
  - @rspress/plugin-auto-nav-sidebar@0.0.6
  - @rspress/plugin-last-updated@0.0.6
  - @rspress/plugin-medium-zoom@0.0.6
  - @rspress/remark-container@0.0.5
  - @rspress/shared@0.0.6

## 0.0.5

### Patch Changes

- 1171bc9: fix: logger params optional

  fix: logger 参数变为可选

- Updated dependencies [1171bc9]
  - @rspress/plugin-auto-nav-sidebar@0.0.5
  - @rspress/plugin-last-updated@0.0.5
  - @rspress/plugin-medium-zoom@0.0.5
  - @rspress/remark-container@0.0.4
  - @rspress/shared@0.0.5

## 0.0.4

### Patch Changes

- e097820: feat: bump mdx-rs version
- 0ab0f9f: feat: add @rspress/remark-container

  feat: 添加 @rspress/remark-container

- Updated dependencies [e097820]
- Updated dependencies [0ab0f9f]
  - @rspress/plugin-auto-nav-sidebar@0.0.4
  - @rspress/plugin-last-updated@0.0.4
  - @rspress/plugin-medium-zoom@0.0.4
  - @rspress/remark-container@0.0.3
  - @rspress/shared@0.0.4

## 0.0.3

### Patch Changes

- c9d37de: feat: replace prefix classname keyword to rspress

  feat: 将所有的类名前缀关键字替换为 rspress

- Updated dependencies [c9d37de]
  - @rspress/plugin-auto-nav-sidebar@0.0.3
  - @rspress/plugin-last-updated@0.0.3
  - @rspress/plugin-medium-zoom@0.0.3
  - @rspress/shared@0.0.3

## 0.0.2

### Patch Changes

- 269d0a9: chore: add docs package
- Updated dependencies [269d0a9]
  - @rspress/plugin-auto-nav-sidebar@0.0.2
  - @rspress/plugin-last-updated@0.0.2
  - @rspress/plugin-medium-zoom@0.0.2
  - @rspress/shared@0.0.2

## 0.0.1

### Patch Changes

- db2d3cd: feat: add plugins
- Updated dependencies [db2d3cd]
  - @rspress/plugin-auto-nav-sidebar@0.0.1
  - @rspress/plugin-last-updated@0.0.1
  - @rspress/plugin-medium-zoom@0.0.1
  - @rspress/shared@0.0.1
