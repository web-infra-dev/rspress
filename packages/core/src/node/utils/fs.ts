import fs from 'node:fs/promises';

export async function pathExists(path: string): Promise<boolean> {
  try {
    await fs.access(path);
    return true;
  } catch {
    return false;
  }
}

export async function readJson(path: string): Promise<any> {
  const raw = await fs.readFile(path, 'utf8');
  return JSON.parse(raw);
}
