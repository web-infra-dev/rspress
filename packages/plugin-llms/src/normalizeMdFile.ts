import {
  type RouteService,
  remarkFileCodeBlock,
  remarkLink,
} from '@rspress/core';
import type { Root } from 'hast';
import remarkMdx from 'remark-mdx';
import remarkParse from 'remark-parse';
import remarkStringify from 'remark-stringify';
import { type Plugin, unified } from 'unified';
import { SKIP, visit } from 'unist-util-visit';
import type { VFile } from 'vfile';

// TODO: currently skip mdxElement, expose the remarkPlugin to users
const mdxToMdPlugin: Plugin<[], Root> = () => {
  return tree => {
    // 1. remove all the import
    visit(tree, 'mdxjsEsm', node => {
      if (node.data?.estree?.body[0].type === 'ImportDeclaration') {
        node.value = '';
        return SKIP;
      }
    });

    // 2. mdxJsxFlowElement -> normal element
    visit(tree, 'mdxJsxFlowElement', (node, index, parent) => {
      if (parent && index !== undefined && Array.isArray(parent.children)) {
        parent.children.splice(index, 1, ...node.children);
        return index - 1;
      }
    });

    visit(tree, 'mdxJsxTextElement', (node, index, parent) => {
      if (parent && index !== undefined && Array.isArray(parent.children)) {
        parent.children.splice(index, 1, ...node.children);
        return index - 1;
      }
    });
  };
};

function noopPlugin() {}

function normalizeMdFile(
  content: string,
  filepath: string,
  routeService: RouteService,
  base: string,
  mdxToMd: boolean,
  isMd: boolean,
): Promise<VFile> {
  return unified()
    .use(remarkParse)
    .use(isMd ? noopPlugin : remarkMdx)
    .use(remarkFileCodeBlock, { filepath })
    .use(!isMd && mdxToMd ? mdxToMdPlugin : noopPlugin)
    .use(remarkLink, {
      cleanUrls: '.md',
      routeService,
      remarkLinkOptions: {
        checkDeadLinks: false,
        autoPrefix: true,
      },
      __base: base,
    } satisfies Parameters<typeof remarkLink>[0])
    .use(remarkStringify)
    .process({
      value: content,
      path: filepath,
    });
}

export { normalizeMdFile };
