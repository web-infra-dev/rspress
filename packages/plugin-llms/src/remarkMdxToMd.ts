import type { Root } from 'mdast';
import type { Plugin } from 'unified';
import { SKIP, visit } from 'unist-util-visit';

type RemarkMdxToMdOptions = {
  /** Callback receiving the gathered virtual JS content. */
  onVirtualFile?: (code: string) => void;
  /**
   * Persist the virtual JS code on the vfile via `file.data.mdxVirtualJs`.
   * Enabled by default so downstream plugins can consume it.
   */
  emitToVFile?: boolean;
};

const DEFAULT_OPTIONS: RemarkMdxToMdOptions = {
  emitToVFile: true,
};

const remarkMdxToMd: Plugin<[RemarkMdxToMdOptions?], Root> =
  (options = {}) =>
  (tree, file) => {
    const { onVirtualFile, emitToVFile } = {
      ...DEFAULT_OPTIONS,
      ...options,
    };

    const originalSource =
      typeof file.value === 'string' ? (file.value as string) : '';
    const mdxSnippets: string[] = [];

    const collectSource = (node: {
      value?: string;
      position?: {
        start?: { offset?: number };
        end?: { offset?: number };
      };
    }): string | null => {
      if (typeof node.value === 'string' && node.value.trim() !== '') {
        return node.value;
      }

      const start = node.position?.start?.offset;
      const end = node.position?.end?.offset;

      if (
        typeof start === 'number' &&
        typeof end === 'number' &&
        start >= 0 &&
        end >= start &&
        originalSource
      ) {
        return originalSource.slice(start, end);
      }

      return null;
    };

    const removeNode = (
      parent: { children?: unknown[] } | null | undefined,
      index: number | null | undefined,
      replacement?: unknown[],
    ) => {
      if (
        !parent ||
        typeof index !== 'number' ||
        !Array.isArray(parent.children)
      ) {
        return;
      }

      if (replacement?.length) {
        parent.children.splice(index, 1, ...replacement);
      } else {
        parent.children.splice(index, 1);
      }
    };

    visit(tree, 'mdxjsEsm', (node: any, index?: number, parent?: any) => {
      const source = collectSource(node);
      if (source) {
        mdxSnippets.push(source.trimEnd());
      }

      if (parent && typeof index === 'number') {
        removeNode(parent, index);
      } else {
        node.value = '';
      }

      return SKIP;
    });

    const handleJsx = (type: 'mdxJsxFlowElement' | 'mdxJsxTextElement') => {
      visit(tree, type, (node: any, index?: number, parent?: any) => {
        const source = collectSource(node);
        if (source) {
          mdxSnippets.push(source.trimEnd());
        }

        // derive a component name for placeholder
        let compName = 'Component';
        if (node && typeof node === 'object') {
          if (typeof node.name === 'string') compName = node.name;
          else if (node?.tagName && typeof node.tagName === 'string')
            compName = node.tagName;
          else if (node?.name && typeof node.name === 'object')
            compName = node.name.name ?? node.name.value ?? compName;
        }

        const placeholder = { type: 'html', value: `<!--${compName} ->` };

        if (
          parent &&
          typeof index === 'number' &&
          Array.isArray(parent.children)
        ) {
          // replace the node with a single html placeholder comment
          parent.children.splice(index, 1, placeholder);
          return SKIP;
        }

        return SKIP;
      });
    };

    handleJsx('mdxJsxFlowElement');
    handleJsx('mdxJsxTextElement');

    const virtualJs = mdxSnippets.join('\n\n').trim();

    if (virtualJs) {
      const finalCode = `${virtualJs}\n`;
      if (emitToVFile) {
        // eslint-disable-next-line no-param-reassign
        (file.data ??= {}).mdxVirtualJs = finalCode;
      }
      onVirtualFile?.(finalCode);
    }
  };

export { remarkMdxToMd, type RemarkMdxToMdOptions };
