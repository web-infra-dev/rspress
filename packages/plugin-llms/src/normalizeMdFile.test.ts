import { readFile } from 'node:fs/promises';
import path from 'node:path';
import { RouteService } from '@rspress/core';
import { describe, expect, it } from 'vitest';
import { normalizeMdFile } from './normalizeMdFile';
import { remarkMdxToMd } from './remarkMdxToMd';

describe('normalizeMdFile', () => {
  it('compiles docs-basic fixture to markdown', async () => {
    const docsRoot = path.join(__dirname, 'fixtures/docs-basic');
    const filePath = path.join(docsRoot, 'index.mdx');
    const content = await readFile(filePath, 'utf-8');

    const routeService = await RouteService.create({
      config: {
        route: {
          extensions: ['.mdx', '.md'],
        },
        lang: 'en',
      },
      externalPages: [],
      scanDir: docsRoot,
    });

    const result = await normalizeMdFile(
      content,
      filePath,
      routeService,
      '/',
      false,
      false,
      [],
    );

    expect(result).toMatchSnapshot();
  });

  it('compiles mdx-to-md', async () => {
    const docsRoot = path.join(__dirname, 'fixtures/docs-basic');
    const filePath = path.join(docsRoot, 'index.mdx');
    const content = await readFile(filePath, 'utf-8');

    const routeService = await RouteService.create({
      config: {
        route: {
          extensions: ['.mdx', '.md'],
        },
        lang: 'en',
      },
      externalPages: [],
      scanDir: docsRoot,
    });

    const result = await normalizeMdFile(
      content,
      filePath,
      routeService,
      '/',
      false,
      false,
      [
        [
          remarkMdxToMd,
          {
            onVirtualFile(code) {
              console.log(code, 222222);
            },
          },
        ],
      ],
    );

    expect(result).toMatchSnapshot();
  });
});
