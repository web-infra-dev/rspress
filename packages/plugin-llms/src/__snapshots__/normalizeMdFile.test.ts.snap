// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`normalizeMdFile > compiles docs-basic fixture to markdown 1`] = `
"# Index

## link

[Link](/foo.md)
[](/llms.txt)

## FileCodeBlock

\`\`\`tsx file="./foo.mdx"
# Foo

\`\`\`

## Mdx

import { List, Table } from '../components/index';

<List items={['foo', 'bar']} />

<Table>
  <tr>
    <td>foo</td>
  </tr>

  <tr>
    <td>bar</td>
  </tr>
</Table>
"
`;
